clang/lib/Basic/Module.cpp

```diff
@@ -303,6 +303,10 @@ bool Module::directlyUses(const Module *Requested) {
   if (!Requested->Parent && Requested->Name == "_Builtin_stddef_max_align_t")
     return true;
 
+  // Anyone is allowed to use our builtin ptrauth.h and its accompanying module.
+  if (!Requested->Parent && Requested->Name == "ptrauth")
```

Comment:
While I haven't understood the patch fully, I believe hardcoding the module name in such a style is not good.

Reply:
Unfortunately this is needed, to match the (compiler-interface-defining) `ptrauth.h` header file we vend.  (specifically this is to allow usage of `ptrauth.h` from `no_undeclared_interfaces` modules, see ed84df008f609.

Reply:
It’s needed in this patch because this patch introduces the `ptrauth.h` header file, which wouldn’t be usable from modules without this change.

If you have an alternative fix I’m open to suggestions.

Reply:
On the one hand, I don't think it is necessary to support modules in the first step. On the other hand, we can try to get the same semantics by adding new mechanism to modules. e.g., we can declare some modules are always directly usable in the module map.